src/autopas/containers/directSum/DirectSum.h

```diff
@@ -107,6 +109,77 @@ class DirectSum : public ParticleContainer<Particle, ParticleCell> {
     f->SoAExtractor((*getHaloCell()), (*getHaloCell())._particleSoABuffer);
   }
 
+  /**
+   * Function to iterate over all pairs of particles in an structure of arrays setting on the gpu
+   * @tparam ParticleFunctor
+   * @tparam Traversal
+   * @param f functor that describes the pair-potential
+   * @param traversal the traversal that will be used
+   * @param useNewton3 whether newton 3 optimization should be used
+   */
+  template <class ParticleFunctor, class Traversal>
+  void iteratePairwiseSoACuda(ParticleFunctor *f, Traversal *traversal, bool useNewton3 = false) {
```

Comment:
this function looks as if it is not needed.
=> user iteratePairwise instead

Reply:
also remove iteratePairwiseSoA

Reply:
=> you will need to adapt AutoTuner<Particle, ParticleCell>::iteratePairwiseTemplateHelper(PairwiseFunctor *f) to use iteratePairwise everywhere